### PR TITLE
Fix the defective checks of APIs

### PR DESCRIPTION
**Pull request by PeiweiHu**
##### Checklist

- [ ] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

File: engines/e_loader_attic.c [hunk 1]
```diff
@@ -70,8 +70,8 @@ static char *file_get_pass(const UI_METHOD *ui_method, char *pass,
     if ((prompt = UI_construct_prompt(ui, desc, info)) == NULL) {
         ATTICerr(0, ERR_R_MALLOC_FAILURE);
         pass = NULL;
-    } else if (!UI_add_input_string(ui, prompt, UI_INPUT_FLAG_DEFAULT_PWD,
-                                    pass, 0, maxsize - 1)) {
+    } else if (UI_add_input_string(ui, prompt, UI_INPUT_FLAG_DEFAULT_PWD,
+                                    pass, 0, maxsize - 1) < 0) {
```
Comment (t8m) [inline engines/e_loader_attic.c:74]: `<= 0` here please

File: test/drbgtest.c [hunk 2]
```diff
@@ -531,7 +531,7 @@ static int test_rand_fork_safety(int i)
         success = 0;
 
     /* request a single byte from each of the DRBGs before the next run */
-    if (!TEST_true(RAND_bytes(random, 1) && RAND_priv_bytes(random, 1)))
+    if (!TEST_true(RAND_bytes(random, 1) > 0 && RAND_priv_bytes(random, 1)))
```
Comment (t8m) [inline test/drbgtest.c:534]: Why not `!TEST_int_gt(RAND_bytes(random, 1), 0) || !TEST_int_gt(RAND_priv_bytes(random, 1), 0)`?

File: test/uitest.c [hunk 3]
```diff
@@ -44,8 +44,8 @@ static int test_old(void)
     /* The wrapper passes the UI userdata as the callback userdata param */
     UI_add_user_data(ui, defpass);
 
-    if (!UI_add_input_string(ui, "prompt", UI_INPUT_FLAG_DEFAULT_PWD,
-                             pass, 0, sizeof(pass) - 1))
+    if (UI_add_input_string(ui, "prompt", UI_INPUT_FLAG_DEFAULT_PWD,
+                             pass, 0, sizeof(pass) - 1) < 0)
```
Comment (t8m) [inline test/uitest.c:48]: `<= 0` please

**Review by t8m — CHANGES_REQUESTED** [review]

File: engines/e_loader_attic.c (hunk 1 above)
Comment (PeiweiHu) [reply engines/e_loader_attic.c]: Thanks for your review. All faults are fixed. :-)

**Review by t8m — APPROVED** [review]

**Review by paulidale — APPROVED** [review]

**Comment by tmshort** [comment]
Pushed to master and openssl-3.0. Thank you!

**Pull request closed without merging**